wgpu-hal/src/lib.rs

```diff
@@ -803,6 +803,7 @@ impl<A: Api> Clone for TextureBinding<'_, A> {
 pub struct BindGroupEntry {
     pub binding: u32,
     pub resource_index: u32,
+    pub size: u32,
```

Comment:
Is this necessary? The reason why BG entry doesn't have this field today is because the BG layout entry has it already:
```rust
pub count: Option<NonZeroU32>,
```

Reply:
wgpu-level BindLayoutGroupDescriptor stuff is generally unavailable at HAL stage, unfortunately. I mean, it's available in Metal, but that's because Metal's bind group layouts are fake.

An alternative, I guess, would be to add count: Vec<u32> to wgpu_hal::vulkan::BindGroupLayout and hack around it in Metal. However, I felt that it's quite easy to pass in from frontend, will likely be useful to practically every HAL implementation, and the alternative is all of them gotta implement their own custom reflection for BindGroupLayout type.

Reply:
It's hard to draw the line between what redundant data to provide and what not. I think the line is drawn rather well right now: it provides almost minimum information at BG creation on top of BGL creation. For example, we aren't redundantly providing `BindingType`, just in case, or the shader stages.
It's not just Metal that remembers stuff from BGL creation and uses it for BG. Vulkan currently keeps the types:
```rust
pub struct BindGroupLayout {
    raw: vk::DescriptorSetLayout,
    desc_count: gpu_descriptor::DescriptorTotalCount,
    types: Vec<vk::DescriptorType>,
}
```

So Vulkan may just turn it into `Vec<(vk::DescriptorType, u32)>` or something similar.

Reply:
Fair enough.